compiler/cgen.nim

```diff
@@ -697,8 +697,12 @@ proc closureSetup(p: BProc, prc: PSym) =
   #echo "created environment: ", env.id, " for ", prc.name.s
   assignLocalVar(p, ls)
   # generate cast assignment:
-  linefmt(p, cpsStmts, "$1 = ($2) ClE_0;$n",
-          rdLoc(env.loc), getTypeDesc(p.module, env.typ))
+  if usesWriteBarrier(p.config):
```

Comment:
This is either a valid bugfix which deserves a testcase or it's wrong logic and should read `` if p.config.selectedGC == gcGo:``!